Doc/library/functools.rst

```diff
@@ -332,7 +332,7 @@ The :mod:`functools` module defines the following functions:
      False
 
    When called, the generic function dispatches on the type of the first
-   argument::
+   argument, unless it is used on class::
```

Comment:
This should be more clear. For example, "Note: if it is used with instance or class methods, then `self`/`cls` is skipped, and the next argument is used." Also add an example illustrating this.